R/Design-class.R

```diff
@@ -409,110 +409,118 @@ DualResponsesSamplesDesign <- function(eff_model,
+#'
+DualResponsesDesign <- function(eff_model,
                                 data,
                                 ...) {
   start <- TDDesign(data = data, ...)
   .DualResponsesDesign(start,
-    Effmodel = Effmodel,
+    eff_model = eff_model,
     data = data
   )
 }
 
-## ===============================================================================
-
-
-##' Class for the time-to-DLT augmented CRM design
-##'
-##' This class has special requirements for the \code{model} and \code{data}
-##' slots in comparison to the parent class \code{\linkS4class{Design}}:
-##'
-##' @slot model the model to be used, an object of or inheriting from class
-##' \code{\linkS4class{GeneralModel}}, see in particular
-##' \code{\linkS4class{DALogisticLogNormal}} and
-##' \code{\linkS4class{TITELogisticLogNormal}} which make use of the
-##' time-to-DLT data
-##' @slot data what is the dose grid, any previous data, etc., contained
-##' in an object of class \code{\linkS4class{DataDA}}
-##' @slot safetyWindow still to be documented.
-##'
-##' @example examples/design-class-DADesign.R
-##' @export
-##' @keywords classes
+
+# DADesign ----
+
+## class ----
+
+#' `DADesign`
+#'
+#' @description `r lifecycle::badge("stable")`
+#'
+#' This class has special requirements for the \code{model} and \code{data}
+#' slots in comparison to the parent class [`Design`]:
+#'
+#' @slot model (`DADesign`), an object of or inheriting from class
+#' [`GeneralModel`], see in particular
+#' [`DALogisticLogNormal`] and
+#' [`TITELogisticLogNormal`] which make use of the
+#' time-to-DLT data
+#' @slot data what is the dose grid, any previous data, etc., contained
+#' in an object of class [`DataDA`]
+#' @slot safetyWindow still to be documented.
+#'
+#' @aliases DADesign
+#' @export
+#'
 .DADesign <-
   setClass(
     Class = "DADesign",
-    representation(
+    representation = list(
       model = "GeneralModel",
       data = "DataDA",
       safetyWindow = "SafetyWindow"
     ),
-    prototype(
+    prototype = prototype(
       model = .DALogisticLogNormal(),
       nextBest = .NextBestNCRM(),
       data = DataDA(doseGrid = 1:2),
       safetyWindow = .SafetyWindowConst()
     ),
-    contains = list("Design")
+    contains = "Design"
   )
-validObject(.DADesign())
-
-
-##' Initialization function for `DADesign`
-##'
-##' @param model see \code{\linkS4class{DADesign}}
-##' @param data see \code{\linkS4class{DADesign}}
-##' @param safetyWindow see \code{\linkS4class{DADesign}}
-##' @param \dots additional arguments for \code{\link{Design}}
-##' @return the \code{\linkS4class{DADesign}} object
-##'
-##' @export
-##' @keywords methods
-DADesign <- function(model,
-                     data,
+
+
+## constructor ----
+
+#' @rdname DADesign-class
+#'
+#' @param model (`DALogisticLogNormal`)\cr see slot definition.
+#' @param data see [`DADesign`]
+#' @param safetyWindow see [`DADesign`]
+#' @param \dots additional arguments for \code{\link{Design}}
+#' @inheritDotParams RuleDesign
```

Comment:
```suggestion
#' @param model (`GeneralModel`)\cr see slot definition.
#' @param data (`DataDA`)\cr see slot definition.
#' @param safetyWindow (`SafetyWindow`)\cr see slot definition.
#' @inheritDotParams Design
```

Reply:
@danielinteractive 
Few questions:
- Am I right that the document for the constructor function of `DADesign` is not meant to be accessible or read by the user, because what they really care is the class?
- If so, why do we add all these details in the header? Is that for the developer? 
- How to read document specifically for the constructor function from rstudio? When I do `?DADesign` in rstudio, I only see the document for the class (composed by the header of .DADesign class)
- What is the difference between ( ) and [ ]?

Reply:
The user constructor (called `DADesign()`) is to be used by the user directly, therefore we export it and document it in the same place. We want that this is in the same place, so when calling `?DADesign` it should all be there - the class documentation and the user constructor.

For the markdown, `()` is just parentheses, but `[]` means a link to the documentation page. We use those differently as you can see from the other roxygen2 chunks.